### PR TITLE
support pyqt6 via qtpy

### PR DESCRIPTION
**Pull request by tlambert03**
just tried to go through the readme to set this up, and since I'm on apple silicon I wanted to use pyqt6.  
So this pr would do that via qtpy (which I see you're already using upstream in the event_bus).

I also needed to add ome-types to the requirements, and made a couple changes there to support ome-types v0.4.0 (that's also my lib by the way, and I'm very happy to see some of the work you did here to write out ome xml from the state of the core.  That's something I'd like to support directly in pymmcore-plus, so might ping you on that as well)

Tests are all passing locally, but It's totally fine if you'd rather not merge this (I understand that supporting PyQt6 just for apple silicon, which will likely never drive a scope, is a bit silly... but perhaps generalizing to qtpy has other benefits for you?)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -38,7 +39,13 @@
         "zarr",
         "ome_zarr",
         "pymm_eventserver",
+        "ome-types[lxml]",
```
Comment (tlambert03) [inline setup.py:42]: note, as of v0.4.0, xmlschema is no longer a required dep of ome-types, so if you want to use the validate_xml method (which you do here) then you need to add either the `lxml` extra (recommended) or the `xmlschema` extra

File: eda_plugin/utility/eda_gui.py [hunk 2]
```diff
@@ -22,7 +22,8 @@
 log = logging.getLogger("EDA")
 
 # Adjust for different screen sizes
-QtWidgets.QApplication.setAttribute(QtCore.Qt.HighDpiScaleFactorRoundingPolicy.PassThrough)
+QtWidgets.QApplication.setHighDpiScaleFactorRoundingPolicy(QtCore.Qt.HighDpiScaleFactorRoundingPolicy.PassThrough)
```
Comment (tlambert03) [inline eda_plugin/utility/eda_gui.py:25]: note:
this is the one thing I couldn't go back and check on pyqt5... but it does look like this is the API that was there in both 5 and 6: https://doc.qt.io/qt-5/qguiapplication.html#setHighDpiScaleFactorRoundingPolicy

File: eda_plugin/utility/ome_metadata.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 import json
 from typing import List
 from pymm_eventserver.data_structures import MMSettings, PyImage, MMChannel
+from ome_types.model import simple_types
```
Comment (tlambert03) [inline eda_plugin/utility/ome_metadata.py:8]: this change will work in both v0.4.0 and below, but the previous code wouldn't

File: setup.py [hunk 4]
```diff
@@ -38,7 +39,13 @@
...
     ],
+    extras_require={
+        'pyqt5': ['PyQt5'],
+        'pyqt6': ['PyQt6'],
+        'test': ['pytest', 'pytest-qt'],
```
Comment (tlambert03) [inline setup.py:47]: if you're so inclined... this is what I needed to get tests working

File: eda_plugin/utility/eda_gui.py (hunk 2 above)
Comment (wl-stepp) [reply eda_plugin/utility/eda_gui.py:25]: I think I saw recently that this setting is now standard in PyQt6

**Comment by wl-stepp** [comment]
Thanks @tlambert03. Will have a closer look tomorrow.

**Pull request merged**